bundles/org.palladiosimulator.monitorrepository.editor/src/org/palladiosimulator/monitorrepository/presentation/MonitorRepositoryEditor.java

```diff
@@ -1,1818 +1,36 @@
-     * 
-     * @generated
-     */
-    private static String getString(final String key, final Object s1) {
-        return MonitorrepositoryEditorPlugin.INSTANCE.getString(key, new Object[] { s1 });
-    }
-
-    /**
-     * This implements {@link org.eclipse.jface.action.IMenuListener} to help fill the context menus
-     * with contributions from the Edit menu. <!-- begin-user-doc --> <!-- end-user-doc -->
-     * 
-     * @generated
-     */
-    @Override
-    public void menuAboutToShow(final IMenuManager menuManager) {
-        ((IMenuListener) this.getEditorSite().getActionBarContributor()).menuAboutToShow(menuManager);
-    }
-
-    /**
-     * <!-- begin-user-doc --> <!-- end-user-doc -->
-     * 
-     * @generated
-     */
-    public EditingDomainActionBarContributor getActionBarContributor() {
-        return (EditingDomainActionBarContributor) this.getEditorSite().getActionBarContributor();
-    }
-
-    /**
-     * <!-- begin-user-doc --> <!-- end-user-doc -->
-     * 
-     * @generated
-     */
-    public IActionBars getActionBars() {
-        return this.getActionBarContributor().getActionBars();
-    }
-
-    /**
-     * <!-- begin-user-doc --> <!-- end-user-doc -->
-     * 
-     * @generated
-     */
-    public AdapterFactory getAdapterFactory() {
-        return this.adapterFactory;
-    }
-
-    /**
-     * <!-- begin-user-doc --> <!-- end-user-doc -->
-     * 
-     * @generated
-     */
-    @Override
-    public void dispose() {
-        this.updateProblemIndication = false;
-
-        ResourcesPlugin.getWorkspace().removeResourceChangeListener(this.resourceChangeListener);
-
-        this.getSite().getPage().removePartListener(this.partListener);
-
-        this.adapterFactory.dispose();
-
-        if (this.getActionBarContributor().getActiveEditor() == this) {
-            this.getActionBarContributor().setActiveEditor(null);
-        }
-
-        for (final PropertySheetPage propertySheetPage : this.propertySheetPages) {
-            propertySheetPage.dispose();
-        }
-
-        if (this.contentOutlinePage != null) {
-            this.contentOutlinePage.dispose();
-        }
-
-        super.dispose();
-    }
-
-    /**
-     * Returns whether the outline view should be presented to the user. <!-- begin-user-doc -->
-     * <!-- end-user-doc -->
-     * 
-     * @generated
-     */
-    protected boolean showOutlineView() {
-        return true;
-    }
-}
+package org.palladiosimulator.monitorrepository.presentation;
+
+import java.io.IOException;
+import java.util.HashMap;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+
+/**
+ * Customized version of {@link MonitorRepositoryEditorGen}.
+ */
+public class MonitorRepositoryEditor extends MonitorRepositoryEditorGen {
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void createModel() {
+		super.createModel();
+		this.addExtraResource("pathmap://METRIC_SPEC_MODELS/models/commonMetrics.metricspec");
```

Comment:
You could replace that magic value by a reference to MetricDescriptionConstants.